app/lib/getNearbyServices.js

```diff
@@ -1,27 +1,53 @@
 const log = require('../../app/lib/logger');
-const docDB = require('../../config/config').docDB;
-const queries = require('../lib/queries');
-const filterServices = require('../lib/filterServices');
-
-function getNearbyServices(req, searchPoint, limits, next) {
-  const geoSpatialQuery = queries.geoSpatialQuery(searchPoint, limits.searchRadius);
-
-  req.app.locals.docDBClient
-    .queryDocuments(docDB.collectionUrl, geoSpatialQuery)
-    .toArray((err, results) => {
-      if (err) {
-        log.error({ err }, 'DocumentDB query failed');
-        next(err);
-      } else {
-        log.debug(`${results.length} RESULTS returned...`);
-
-        log.debug('filter-and-sort-services-start');
-        const filteredServices = filterServices(results, limits);
-        log.debug('filter-and-sort-services-end');
+const MongoClient = require('mongodb').MongoClient;
+const co = require('co');
+const mongodbConfig = require('../../config/config').mongodb;
+const filterServices = require('./filterServices');
+const VError = require('verror').VError;
 
-        next(null, filteredServices);
+const connectionString = mongodbConfig.connectionString;
+
+function getNearbyServices(searchPoint, limits, next) {
+  co(function* a() {
+    const db = yield MongoClient.connect(connectionString);
+    log.debug(`Connected to ${mongodbConfig.connectionString}`);
+
+    const col = db.collection(mongodbConfig.collection);
+
+    col.aggregate([{
+      $geoNear: {
+        near: { type: 'Point', coordinates: searchPoint.coordinates },
+        distanceField: 'dist',
+        maxDistance: 32180,
+        distanceMultiplier: 0.001, // this is in miles
+        num: 2500, // Arbitary number of results to make sure we get everything within 20 miles
```

Comment:
What happens if 2500 is increased to 5000? Does the response time increase by much/anything? And how does that work in collaboration with changing `maxDistance` or even removing that?